apps/web/src/components/wizard/index.tsx

```diff
@@ -18,27 +18,34 @@ import { uploadChallenge } from './create.action';
 import DEFAULT_CHALLENGE_TEMPLATE from './default-challenge.md';
 import DEFAULT_TEST_CASES from './default-tests.md';
 import DEFAULT_DESCRIPTION from './default-description.md';
+import { containsProfanity } from '~/utils/profanity';
 
 export const enum STEPS {
   ChallengeCard,
   Description,
   TestCases,
   Summary,
 }
+
+
 const testCaseRegex = new RegExp('(?:\n|^)s*(?:Equal|Extends|NotEqual|Expect)<');
 const createExploreCardSchema = z.object({
   difficulty: z.enum(['BEGINNER', 'EASY', 'MEDIUM', 'HARD', 'EXTREME']),
   name: z
     .string()
     .min(3, 'The name must be longer than 3 characters')
-    .max(30, 'The name must be shorter than 30 characters'),
+    .max(30, 'The name must be shorter than 30 characters')
+    .refine((str: string) => !containsProfanity(str), "The name must not contain bad words."),
```

Comment:
we can probably just make a custom zod type so you dont have to repeat this

Reply:
it's really only used here (I think), and I'm working on a textarea that prevents submission if there's profanity, so it might not be necessary

I'll see how it goes with the textarea

Reply:
this needs to be applied to any place youre allowed to use text. comments, solutions, challenge submissions